### PR TITLE
Remove asserts and replace with GDTLogs

### PR DESCRIPTION
**Pull request by mikehaney24**
Should infrequently log to the console in !NDEBUG builds instead of asserting.
Fixes 3530

### CODE REVIEW

File: GoogleDataTransport/GDTLibrary/GDTConsoleLogger.m [hunk 1]
```diff
@@ -14,7 +14,7 @@
  * limitations under the License.
  */
 
-#import "GDTLibrary/Private/GDTConsoleLogger.h"
+#import "GDTLibrary/Public/GDTConsoleLogger.h"
```
Comment (paulb777) [inline GoogleDataTransport/GDTLibrary/GDTConsoleLogger.m]: Why does it need to be public now?

Comment (mikehaney24) [reply GoogleDataTransport/GDTLibrary/GDTConsoleLogger.m]: Because it'll be used in libraries that depend on this library (GDTCCT and FIRCD, specifically)

**Comment by paulb777** [comment]
Looks like three libraries will need to be updated. Minor update to GDT and patches to the other 2. Should we tag master after these land or cherry-pick to the release-6.6.0 branch?  Also the three podspecs need updates.

**Comment by mikehaney24** [comment]
Restoring to create another PR onto release-6.6.0

**Pull request merged**